packages/doenetml-worker-rust/lib-doenetml-core/src/core/state_var_interfaces/text_state_var_interfaces.rs

```diff
@@ -0,0 +1,197 @@
+                    string_vals.push(dep_string_value.create_new_read_only_view())
+                } else {
+                    panic!("Got a non-string value for a dependency for a GeneralStringStateVarInterface");
+                }
+            }
+        }
+
+        self.string_dependency_values = string_vals;
+
+        if num_dependencies == 1 {
+            if let DependencySource::Essential { .. } = dependencies[0][0].source {
+                self.from_single_essential = true;
+            }
+        }
+    }
+
+    fn calculate_state_var_from_dependencies_and_mark_fresh(
+        &self,
+        state_var: &StateVarMutableViewTyped<String>,
+    ) {
+        if self.from_single_essential {
+            // if we are basing it on a single essential variable,
+            // then we propagate used_default as well as the value.
+            state_var.set_value_and_used_default(
+                self.string_dependency_values[0].get_fresh_value().clone(),
+                self.string_dependency_values[0].get_used_default(),
+            );
+        } else {
+            // TODO: can we implement this without cloning the inner value?
+            let value: String = self
+                .string_dependency_values
+                .iter()
+                .map(|v| v.get_fresh_value().clone())
+                .collect();
+
+            state_var.set_value(value);
+        }
+    }
+
+    fn request_dependencies_to_update_value(
+        &self,
+        state_var: &StateVarReadOnlyViewTyped<String>,
+        _is_direct_change_from_renderer: bool,
+    ) -> Result<Vec<DependencyValueUpdateRequest>, RequestDependencyUpdateError> {
+        if self.string_dependency_values.len() != 1 {
+            // TODO: implement for no dependencies where saves to essential value?
+            Err(RequestDependencyUpdateError::CouldNotUpdate)
+        } else {
+            let desired_value = state_var.get_requested_value();
+
+            self.string_dependency_values[0].request_change_value_to(desired_value.clone());
+
+            Ok(vec![DependencyValueUpdateRequest {
+                instruction_idx: 0,
+                dependency_idx: 0,
+            }])
+        }
+    }
+}
+
+/// A simplified version of GeneralStringStateVarInterface
+/// that is based on a single dependency.
+/// Requires a `dependency_instruction_hint`.
+///
+/// Panics if `create_dependency_from_extend_source` is true,
+/// if not given a `dependency_instruction_hint`,
+/// or if given a `dependency_instruction_hint`
+/// that doesn't result in at least one string dependency.
+#[derive(Debug, Default)]
+pub struct SingleDependencyStringStateVarInterface {
+    string_dependency_value: StateVarReadOnlyViewTyped<String>,
+}
+
+impl StateVarInterface<String> for SingleDependencyStringStateVarInterface {
+    fn return_dependency_instructions(
+        &self,
+        _extend_source: Option<&ExtendSource>,
```

Comment:
I feel like `extending` or `extending_from` would be easier to understand.

Reply:
Yeah, I've been meaning to go find my notes where we discussed better options for this terminology.

Reply:
It looks like our last discussion ended with the proposal:
```
<point name="P>(1,2)</point>
<point extend="$P" x="3" />
<point clone="$P" x="3" />
```
So we'd still be using the word "extend". For "clone", there would be no link so nothing to add to the data structure.

You are suggesting we change the function parameter name. Are you also suggesting we change the name of the `enum` to something other than `ExtendSource`? I haven't come up with a better name for that. I'm OK with changing the function parameter name to either of the suggestions.

Reply:
I was suggesting just the parameter name.

To me `<point name="Q" extend="$P" />` reads as `the point $Q extends the component $P`. But when you ask a question `what does $Q extend?`, I prefer `$Q is extending $P`, rather than `the source that $Q declares it will extend is $P$`.